src/open_inwoner/userfeed/hooks/external_task.py

```diff
@@ -0,0 +1,88 @@
+from typing import List
+
+from django.utils.translation import gettext_lazy as _
+
+from open_inwoner.accounts.choices import LoginTypeChoices
+from open_inwoner.accounts.models import User
+from open_inwoner.openzaak.api_models import OpenTask
+from open_inwoner.openzaak.clients import build_client
+from open_inwoner.userfeed.adapter import FeedItem
+from open_inwoner.userfeed.choices import FeedItemType
+from open_inwoner.userfeed.models import FeedItemData
+
+from ..adapters import register_item_adapter
+
+
+class OpenTaskFeedItem(FeedItem):
+    base_title = _("Open task")
+    base_message = _("Open task that is yet to be completed")
+
+    @property
+    def title(self) -> str:
+        return f"{self.base_title} ({self.get_data('task_identificatie')})"
+
+    @property
+    def message(self) -> str:
+        return self.get_data("task_name", super().message)
+
+
+def update_external_task_items(user: User, openstaande_taken: List[OpenTask]):
+    """
+    Creates items for OpenTasks if they do not exist yet, updates existing items if the
+    data changed and marks existing items as complete if no OpenTask exists for that
+    uuid anymore
+    """
+    existing_uuid_mapping = {
+        str(item.ref_uuid): item
+        for item in FeedItemData.objects.filter(
+            type=FeedItemType.external_task,
+            user=user,
+        )
+    }
+    existing_uuids = set(existing_uuid_mapping.keys())
+
+    update_data = []
+    create_data = []
+    for task in openstaande_taken:
+        type_data = dict(
+            action_url=task.formulier_link,
+            task_name=task.naam,
+            task_identificatie=task.identificatie,
+        )
+        if existing_item := existing_uuid_mapping.get(task.uuid):
+            if existing_item.type_data != type_data:
+                existing_item.type_data = type_data
+                update_data.append(existing_item)
+            continue
+
+        data = {
+            "user": user,
+            "type": FeedItemType.external_task,
+            "ref_uuid": task.uuid,
+            "action_required": True,
+            "type_data": dict(
+                action_url=task.formulier_link,
+                task_name=task.naam,
+                task_identificatie=task.identificatie,
+            ),
+        }
+        create_data.append(FeedItemData(**data))
```

Comment:
.. but this should be a FeedItemData() constructor so the IDE can type check. Using an intermediate dict literal and **kwargs to put them in the constructor is unnecessary.  (this is opposite from earlier comment about the type_data)

Reply:
oops I did it the wrong way around, the type_data is now a regular dict (was also duplicated before, so I removed the redeclaration) and i use the constructor with keyword args for FeedItemData

Reply:
Cool, thanks. Much clearer.